src/common/camera_control.c

```diff
@@ -874,7 +876,23 @@ gboolean _camera_initialize(const dt_camctl_t *c, dt_camera_t *cam)
   return TRUE;
 }
 
-
+static time_t _get_exif_time_from_data(uint8_t* data, unsigned long size)
+{
+  dt_image_t img;
+  dt_exif_read_from_blob(&img, data, size);
```

Comment:
Don't use that, it has side effects, like writing stuff to database. Instead add something like `dt_exif_get_datetime()` to `common/exif.cc` which does things similar to `dt_exif_get_color_space()`.